packages/core/src/integrations/requestdata.ts

```diff
@@ -73,15 +74,24 @@ const _requestDataIntegration = ((options: RequestDataIntegrationOptions = {}) =
       // that's happened, it will be easier to add this logic in without worrying about unexpected side effects.)
 
       const { sdkProcessingMetadata = {} } = event;
-      const req = sdkProcessingMetadata.request;
+      const { request, normalizedRequest } = sdkProcessingMetadata;
 
-      if (!req) {
-        return event;
+      const addRequestDataOptions = convertReqDataIntegrationOptsToAddReqDataOpts(_options);
+
+      // If this is set, it takes precedence over the plain request object
+      if (normalizedRequest) {
+        // Some other data is not available in standard HTTP requests, but can sometimes be augmented by e.g. Express or Next.js
+        const ipAddress = request ? request.ip || (request.socket && request.socket.remoteAddress) : undefined;
+        const user = request ? request.user : undefined;
+
+        return addNormalizedRequestDataToEvent(event, normalizedRequest, { ipAddress, user }, addRequestDataOptions);
       }
 
-      const addRequestDataOptions = convertReqDataIntegrationOptsToAddReqDataOpts(_options);
+      if (!request) {
+        return event;
+      }
 
-      return addRequestDataToEvent(event, req, addRequestDataOptions);
+      return addRequestDataToEvent(event, request, addRequestDataOptions);
```

Comment:
I would mayyyybee do it like this?:


```suggestion
      const { sdkProcessingMetadata = {} } = event;
      const { request, normalizedRequest } = sdkProcessingMetadata;

      const addRequestDataOptions = convertReqDataIntegrationOptsToAddReqDataOpts(_options);

      if (request) {
        addRequestDataToEvent(event, request, addRequestDataOptions);
      }

      // If this is set, it takes precedence over the plain request object
      if (normalizedRequest) {
        addNormalizedRequestDataToEvent(event, normalizedRequest, addRequestDataOptions);
      }

      return event;
```

So that
- We always use all of the data.
- For now we can just use the old logic from `addRequestDataToEvent` to extract the "non-standard" fields on the request. Removing the need for the `additionalData` arg which I admittedly find a bit offputting.

Reply:
(Sorry, took some time to come back to this 😅 )

I guess my main problem with this is that we do all the work of parsing the "old" request etc. even though we don't need it? I do feel like it would be easier to understand to have a single function be responsible for it - otherwise, we end up in the state of "why is this field set here? let's look through all of the old code to find if it sets it somewhere, then look over the new code, ...". 

By not calling the old method anymore it makes it easier to follow the "new, happy" path imho 🤔 though I do agree that the additionalData part is not great...

Reply:
@lforst ? 🙃

Reply:
@maximedupre we will get to it. Please understand that we are bound to other obligations, absences, deadlines... Pinging usually doesn't influence in what order we do things so I would generally just not do it - in any open source project. Thanks!